src/components/SearchForm/SearchForm.test.js

```diff
@@ -0,0 +1,32 @@
+import { screen } from '@testing-library/react';
+import SearchForm from '.';
+
+
+describe('SearchForm', () => {
+    beforeEach(() => render(<SearchForm/>))
+    
+    test('it renders a button', () => {
+        expect(screen.getByRole('button')).toBeInTheDocument()
+    })
+
+    test('it renders a input', () => {
+        expect(screen.getByRole('textbox')).toBeInTheDocument()
+    })
+
+    test('it renders a form', () => {
+        expect(screen.getByRole('form', {name:'search'})).toBeInTheDocument()
+    })
+    
+    test('it triggers handle submit on submit', () => {
+
+        const handleSubmit = jest.fn()
+        const form= screen.getByRole('form')
+        userEvent.submit(form)
+        expect(handleSubmit).toHaveBeenCalledTimes(1)
+
+    // need to work on this
```

Comment:
Good tests, we found this as well, with userEvent.submit is not a function.

Reply:
We fixed this by changing the test, instead of targeting the form and expecting a submit, we targeted the submit button and simulated a click event. We saw you also tried to mock handleSubmit, we aren't sure if this is allowed as it is declared in the file that you are testing. To make this test pass, we first defined a mock function and input that to the component in the beforeEach.`
    const mockFunc = jest.fn()
    beforeEach(() => render(<SearchForm searchForUser={mockFunc}/>))`.
For the test, we then expected this mock function to have been called. `
        const submitButton = screen.getByRole('button');
        userEvent.click(submitButton);
        expect(mockFunc).toHaveBeenCalledTimes(1)`.

To make this work we added role="button" to the submit input in the form so it can be targeted by the getByRole.

Reply:
Thanks for this :D will take a look into making it work